src/functional/iterator/embedded.py

```diff
@@ -586,27 +594,31 @@ def shape(self):
         return self.array().shape
 
 
+def _is_tuple_axis(axis: Axis):
+    return axis is None
+
+
 def get_ordered_indices(
     axes: Iterable[Axis], pos: Mapping[str, FieldIndex]
 ) -> tuple[FieldIndex, ...]:
-    assert all(axis.value in [*pos.keys()] for axis in axes)
-    return tuple(pos[axis.value] for axis in axes)
+    assert all(axis.value in [*pos.keys()] for axis in axes if axis is not None)
```

Comment:
Just an alternative: `Mapping`s' iterators are on keys by default:
```suggestion
    assert all(axis.value in [*pos] for axis in axes if axis is not None)
```

Reply:
Makes sense, for my taste Python has the wrong default here, that's why I unconsciously use the explicit form.